packages/eslint-plugin/src/rules/prefer-find.ts

```diff
@@ -0,0 +1,283 @@
+     */
+    function isTreatedAsZeroByArrayAt(value: unknown): boolean {
+      const asNumber = Number(value);
+
+      if (isNaN(asNumber)) {
+        return true;
+      }
+
+      return Math.trunc(asNumber) === 0;
+    }
+
+    function isMemberAccessOfZero(
+      node: TSESTree.MemberExpressionComputedName,
+    ): boolean {
+      const property = getStaticValue(node.property, globalScope);
+      // Check if it looks like <<stuff>>[0] or <<stuff>>['0'], but not <<stuff>>?.[0]
+      return (
+        !node.optional &&
+        property != null &&
+        isTreatedAsZeroByMemberAccess(property.value)
+      );
+    }
+
+    /**
+     * Implements the algorithm for array indexing by member operator.
+     * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#array_indices
+     */
+    function isTreatedAsZeroByMemberAccess(value: unknown): boolean {
+      return String(value) === '0';
+    }
+
+    function generateFixToRemoveArrayElementAccess(
+      fixer: TSESLint.RuleFixer,
+      arrayNode: TSESTree.Expression,
+      wholeExpressionBeingFlagged: TSESTree.Expression,
+      sourceCode: SourceCode,
+    ): RuleFix {
+      const tokenToStartDeletingFrom = nullThrows(
+        // The next `.` or `[` is what we're looking for.
+        // think of (...).at(0) or (...)[0] or even (...)["at"](0).
+        sourceCode.getTokenAfter(
+          arrayNode,
+          token => token.value === '.' || token.value === '[',
+        ),
+        'Expected to find a member access token!',
+      );
+      return fixer.removeRange([
+        tokenToStartDeletingFrom.range[0],
+        wholeExpressionBeingFlagged.range[1],
+      ]);
+    }
+
+    function generateFixToReplaceFilterWithFind(
+      fixer: TSESLint.RuleFixer,
+      filterExpression: {
+        isBracketSyntaxForFilter: boolean;
+        filterNode: TSESTree.Node;
+      },
+    ): TSESLint.RuleFix {
+      return fixer.replaceText(
+        filterExpression.filterNode,
+        filterExpression.isBracketSyntaxForFilter ? '"find"' : 'find',
+      );
+    }
+
+    return {
+      CallExpression(node): void {
+        const object = getObjectIfArrayAtExpression(node);
+        if (object) {
+          const filterExpression = parseIfArrayFilterExpression(object);
+          if (filterExpression) {
+            context.report({
+              node,
+              messageId: 'preferFind',
+              suggest: [
+                {
+                  messageId: 'preferFindSuggestion',
+                  fix: (fixer): TSESLint.RuleFix[] => {
+                    const sourceCode = getSourceCode(context);
+                    return [
+                      generateFixToReplaceFilterWithFind(
+                        fixer,
+                        filterExpression,
+                      ),
+                      // get rid of the .at(0) or ['at'](0)
```

Comment:
[Nit] The later comment capitalizes the 'g'. This one doesn't. Up for switching to always the same style? 😄

Reply:
changed